bin/idsnp_controller-myeolid_specific.pl

```diff
@@ -0,0 +1,226 @@
+#!/usr/bin/env perl
+
+use strict;
+use warnings;
+use Getopt::Long;
+use Data::Dumper;
+use JSON;
+
+our %opt;
+GetOptions( \%opt, 'vcf_sample=s', 'vcf_control=s', 'sample=s', 'control=s', 'rs_bed=s', 'help' );
+my $SCORE   = -20;
+
+my $MARKERS = 0; # originally 44 but has been cut down to 40 as off 2024-02-02 again but cound from the vcf seems more logical
+
+# my $MARKERS = 111; 
+
+print_usage(
+    "ERROR: --vcf_sample, --vcf_control, --sample and --control required!\n")
+  unless $opt{vcf_sample}
+  and $opt{vcf_control}
+  and $opt{sample}
+  and $opt{control}
+  and $opt{rs_bed};
+
+unless ( -e $opt{vcf_sample} ) {
+    print "ERROR: SAMPLE VCF file not found\n";
+    exit;
+}
+
+#unless( -e $opt{vcf_control} ) { print "ERROR: CONTROL VCF file not found\n"; exit; }
+
+if ( defined( $opt{rs_bed} ) ) {
+  open( my $rsbed, '<', "$opt{rs_bed}" ) or die "Could not open file";
+  while (<$rsbed>) {
+    $MARKERS++;
+    }
+  close($rsbed);
+}
+
+
+my %data;
+
+# sample vcf
+if ( defined( $opt{vcf_sample} ) ) {
+
+    if ( $opt{vcf_sample} =~ /gz$/ ) {
+        open( sVCF, "zcat $opt{vcf_sample} |" )
+          or die "gunzip $opt{vcf_sample}: $!";
+    }
+    else {
+        open( sVCF, "$opt{vcf_sample}" ) or die "gunzip $opt{vcf_sample}: $!";
+    }
+    my %labels;
+    while ( my $row = <sVCF> ) {
+        chomp($row);
+        my @cols = split( /\t/, $row );
+        next if ( $row =~ /^##/ );
+        if ( $row =~ /^#CHROM/ ) {
+            for my $index ( 0 .. $#cols ) { $labels{ $cols[$index] } = $index; }
+        }
+        else {
+            my $identifier = "".$cols[ $labels{'#CHROM'} ]."_".$cols[$labels{POS}]."_".$cols[$labels{ID}]."";
+            #my $identifier = "". $cols[ $labels{'#CHROM'} ] . "_". $cols[ $labels{POS} ] . "";
+            my $id = $cols[ $labels{ID} ];
+            my $gt = ( split( /:/, $cols[ $labels{FORMAT} + 1 ] ) )[0];
+            my @info = split( /;/, $cols[ $labels{INFO} ] );
+            my ( $reff, $refr, $altf, $altr );
+            for my $c (@info) {
+                chomp($c);
+                if ( $c =~ /^DP4=(.*)/ ) {
+                    ( $reff, $refr, $altf, $altr ) = split( /,/, $1 );
+                }
+            }
+            $data{$identifier}{ $opt{sample} }{genotype} = $gt;
+            $data{$identifier}{ $opt{sample} }{quality} =
+              $cols[ $labels{QUAL} ];
+            $data{$identifier}{ $opt{sample} }{dp} =
+              $reff + $refr + $altf + $altr;
+            $data{$identifier}{ $opt{sample} }{dp_ref} = $reff + $refr;
+            $data{$identifier}{ $opt{sample} }{dp_alt} = $altf + $altr;
+        }
+    }
+    close sVCF;
+}
+
+# control vcf
+if ( defined( $opt{vcf_control} ) ) {
```

Comment:
could be nice to add and else-statement that exits since control-vcf is actually a required input

Reply:
will make a exceptions to catch it